File: mobile_client/app/src/main/java/protocol/BlindsDOWNOnTimeBuilder.java
package protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class BlindsDOWNOnTimeBuilder implements CommandBuilder, Closeable
	{
		private final int INIT_EPOCH_TIME_POSITION_IN_FRAME = 6;

		public BlindsDOWNOnTimeBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			int dateAndTime = BytesConverter.fromVectorOfUINT8toUINT32(commandInBytes, INIT_EPOCH_TIME_POSITION_IN_FRAME);
			Command command  = new BlindsDOWNOnTimeCommand(dateAndTime);
			return command;
		}
	}

File: server/src/protocol/BlindsDOWNOnTimeBuilder.cpp
#include "BlindsDOWNOnTimeBuilder.h"
#include "BlindsDOWNOnTimeCommand.h"

#include <utility/BytesConverter.h>

using namespace std;
using namespace utility;
using namespace communication;

BlindsDOWNOnTimeBuilder::BlindsDOWNOnTimeBuilder()
{}

BlindsDOWNOnTimeBuilder::~BlindsDOWNOnTimeBuilder()
{}

std::unique_ptr<Command> BlindsDOWNOnTimeBuilder::create(const std::vector<uint8_t> &commandInBytes)
{
    uint32_t epochTime;

    epochTime = BytesConverter::fromVectorOfUINT8toUINT32(commandInBytes, INIT_EPOCH_TIME_POSITION_IN_FRAME);

    auto command = make_unique<BlindsDOWNOnTimeCommand>(epochTime);

    return move(command);
}


